src/views/RoomSelector.vue

```diff
@@ -66,26 +74,12 @@ export default {
 			rooms: [],
 			selectedRoom: null,
 			loading: true,
-			// TODO: should be included once this is properly available
-			types: {
-				ROOM_TYPE_ONE_TO_ONE: 1,
-				ROOM_TYPE_GROUP: 2,
-				ROOM_TYPE_PUBLIC: 3,
-				ROOM_TYPE_CHANGELOG: 4,
-			},
 		}
 	},
 	computed: {
-		currentRoom() {
-			if (OCA.SpreedMe && OCA.SpreedMe.app.activeRoom) {
-				return OCA.SpreedMe.app.activeRoom.get('token')
-			}
-			return null
-		},
 		availableRooms() {
 			return this.rooms.filter((room) => {
-				return room.token !== this.currentRoom
```

Comment:
as per julius this is needed in the collaboration resource dialog. you shouldn't be able to bind the conversation to itself. Maybe add an option?

Reply:
since we decided to move away from the collaboration dialog, this will become possible, yes

Reply:
so want to do this check in combination with `this.showPostableOnly` or need another way?

Reply:
I originally thought you were talking about the read-only case.

Now having a closer look at the mentioned line to answer: the "this.currentRoom" was based on an apparently old namespace "OCA.SpreedMe" which I didn't find elsewhere. So I assumed that this information was not available any more and is outdated. Since "this.currentRoom" relied on "OCA.SpeedMe" I removed it.

I could add it back and accept an optional room token but it this approach wouldn't be used anywhere. So maybe we keep my PR as is and add currentRoom back once the dialog is used in such way ?

Reply:
It is used in the collaboration resource feature where you can (without this check) link a conversation to itself. I guess it is broken since the vue rewrite

Reply:
The reason you can't find it atm is #5105

Reply:
If you can point me at a way to trigger it (if #5105 can bring it back), I can repair it to make it work again.

We might need to resort to similar hacks to detect the context, like if we're inside a conversation, because OCA.Collaboration.trigger() doesn't allow passing any arguments, so we can't set RoomSelector properties from there.

Reply:
#5105 is merged, so now:

1. Create a conversation
2. Open the sidebar in the conversation
3. Click on "Add to a project"
4. Click on "Link to a conversation"
5. See same conversation being offered

Reply:
Thanks, I'll try it out. Cool to finally see what this "add to project" is about.

Should the backend also prevent this "linking to self" ?

Reply:
> Should the backend also prevent this "linking to self" ?

It is (silently) ignoring it already
https://github.com/nextcloud/server/blob/5bf3d1bb384da56adbf205752be8f840aac3b0c5/core/Controller/CollaborationResourcesController.php#L135-L138

Reply:
eww, it seems this thing is even more buggy: you can link to another conversation or file multiple times
![image](https://user-images.githubusercontent.com/277525/107745878-4556f680-6d15-11eb-94c6-b8270ce9053e.png)

something to handle separately, seems that feature needs some love...

I'll now focus only on preventing the regression caused by my PR

Reply:
> eww, it seems this thing is even more buggy: you can link to another conversation or file multiple times

This is expected. because it could be different circles of people and projects. But anyway this is part of server and not Talk